app/src/main/java/com/github/se/eventradar/model/event/EventDetailsViewModel.kt

```diff
@@ -13,76 +15,98 @@ import kotlinx.coroutines.flow.StateFlow
+            }
+          }
+          .addOnFailureListener { exception ->
+            // TODO Handle any errors that occur during fetching
+            Log.d(tag, "Error fetching event: $exception")
+          }
+    } else {
+      Log.d(tag, "id is null")
+    }
+  }
 
-    // Fetch the specific document from Firebase
-    db.collection("EVENT")
-        .document(eventId)
-        .get()
-        .addOnSuccessListener { document ->
-          // Check if the document exists
-          if (document.exists()) {
-            // Extract data from the document and create an Event object
-            val latitude = document.getGeoPoint("location")!!.latitude
-            val longitude = document.getGeoPoint("location")!!.longitude
-            val ticketName = document.getString("Ticket Name") ?: ""
-            val ticketPrice = document.getLong("Ticket Price")!!.toDouble()
-            val ticketQuantity = document.getLong("Ticket Quantity")!!.toInt()
-            val address = document.getString("Address") ?: ""
-            val category = document.getString("Category") ?: ""
-            val startInstant: Instant = document.getTimestamp("Start")!!.toDate().toInstant()
-            val endInstant: Instant = document.getTimestamp("End")!!.toDate().toInstant()
+  private fun fillUIStateFromEvent(event: Event) {
+    _uiState.value =
+        _uiState.value.copy(
+            eventName = event.eventName,
+            eventPhoto = event.eventPhoto,
+            start = event.start,
+            end = event.end,
+            location = event.location,
+            description = event.description,
+            ticket = event.ticket,
+            contact = event.contact,
+            category = event.category,
+        )
+  }
 
-            _uiState.value.eventName = document.getString("Name") ?: ""
-            _uiState.value.eventPhoto = document.getString("Photo") ?: ""
-            _uiState.value.start = LocalDateTime.ofInstant(startInstant, ZoneId.systemDefault())
-            _uiState.value.end = LocalDateTime.ofInstant(endInstant, ZoneId.systemDefault())
-            _uiState.value.location = Location(latitude, longitude, address)
-            _uiState.value.description = document.getString("Description") ?: ""
-            _uiState.value.ticket = Ticket(ticketName, ticketPrice, ticketQuantity)
-            _uiState.value.contact = document.getString("Contact") ?: ""
-            _uiState.value.category = EventCategory.valueOf(category)
-          } else {
-            // Document does not exist
-            println("Document does not exist")
-          }
-        }
-        .addOnFailureListener { exception ->
-          // Handle any errors that occur during fetching
-          // You may want to log the error or show a message to the user
-          println("Error fetching event: $exception")
-        }
+  private fun createEventFromDocument(document: DocumentSnapshot, eventId: String): Event {
```

Comment:
That's a good point. I like to separate things (maybe to much). I am wondering what the Event model will be used for. Because one of my next task is to create the Create event screen and add functions to the viewModel, and naturally later, there will be the Edit event Screen as well...

You're right that there is no need now to have them separate, so I can remove that middle man.